### PR TITLE
[SEDONA-437] Add implicit CRS transformation

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-437. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- All raster functions that take a geometry, should implicitly transform the CRS of the geometry if needed.
- The default CRS for these functions should be 4326 for raster and geometry both.

## How was this patch tested?

- passed existing and new tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

**Comment by jiayuasu** [comment]
@furqaankhan Please address the confligc

File: docs/api/sql/Raster-operators.md [hunk 1]
```diff
@@ -1880,19 +1883,21 @@ Introduction: Returns the values at the given points or grid coordinates in the
...
 `RS_Values (raster: Raster, points: ARRAY[Geometry])`
 
 `RS_Values (raster: Raster, points: ARRAY[Geometry], band: Integer)`
 
-`RS_Values (raster: Raster, xCoordinates: ARRAY[Integer], yCoordinates: ARRAY[Integer], band: Integer)`
+```
+RS_Values (raster: Raster, xCoordinates: ARRAY[Integer], yCoordinates: ARRAY[Integer], band: Integer)
+```
 
 Since: `v1.4.0`
```
Comment (furqaankhan) [inline docs/api/sql/Raster-operators.md:1899]: @jiayuasu should I change the version number here?

Comment (jiayuasu) [reply docs/api/sql/Raster-operators.md:1899]: @furqaankhan No need to change.

Comment (furqaankhan) [reply docs/api/sql/Raster-operators.md:1899]: If I don't then it will update to the sedona.apache.org

**Comment by jiayuasu** [comment]
@Kontinuation Please review. Note that: when either geometry or raster has `srid == 0`, we assume it is in 4326 and set their SRID accordingly.

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**